src/software/ai/navigator/navigating_primitive_creator.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "shared/proto/primitive.pb.h"
+#include "software/ai/intent/all_intents.h"
+#include "software/ai/intent/navigating_intent.h"
+#include "software/ai/navigator/obstacle/obstacle.h"
+#include "software/ai/navigator/path_planner/path_planner.h"
+#include "software/parameter/dynamic_parameters.h"
+#include "software/world/world.h"
+
+/**
+ * The NavigatingPrimitiveCreator converts the given NavigatingIntents into their
+ * respective Primitives given the associated path
+ */
+class NavigatingPrimitiveCreator : public NavigatingIntentVisitor
+{
+   public:
+    /**
+     * Create a NavigatingPrimitiveCreator
+     *
+     * @param config The navigator config
+     */
+    explicit NavigatingPrimitiveCreator(std::shared_ptr<const NavigatorConfig> config);
+
+    /**
+     * Creates a primitive for a given path and navigating intent
+     *
+     * @param intent The NavigatingIntent to make primitive from
+     * @param path path to make primitive for
+     * @param enemy_robot_obstacles list of enemy robot obstacles to watch out for
+     *
+     * @return Primitive
+     */
+    TbotsProto::Primitive createNavigatingPrimitive(
```

Comment:
I was curious about the rationale for having `NavigatingPrimitiveCreator` responsible for velocity around path knots and enemy robots, instead of having the `Navigator` responsible for it and velocity fully encapsulated in `Path`.

Reply:
The idea is that (NavigatingIntent, Path) in and Primitive out should support trajectory planner as well, i.e. Path and NavigatingIntent in and trajectory planner primitive out, so that's why I chose that abstraction. @garethellis0 thoughts?

Reply:
essentially, this is a _temporary_ solution before trajectory planner integration. With the trajectory planner, we'll be sending the entire path to the robots, not just one final speed and one final destination

Reply:
The way I understand this is, both `PathManager` and `NavigatingPrimitiveCreator` will both be responsible for some sort of obstacle avoidance functionality?

Reply:
That's kind of right. `PathManager` creates paths that don't intersect with each other and don't intersect obstacles, but it can't control the speed of the robots, because `Path` can't express speed as it is now. The `NavigatingPrimitiveCreator` directly translates these paths into primitives so that the robot can follow the path, but it additionally slows robots down near enemy robot obstacles to be more safe.

Reply:
Ok, I see now. LGTM